src/main/java/org/radarcns/management/repository/SubjectRepository.java

```diff
@@ -18,10 +19,13 @@
     @Query("select distinct subject from Subject subject left join fetch subject.sources")
     List<Subject> findAllWithEagerRelationships();
 
-    @Query("select distinct subject from Subject subject left join fetch subject.sources where subject.user.project.id = :projectId")
+    @Query("select distinct subject from Subject subject left join fetch subject.sources "
+        + "left join fetch subject.user user "
+        + "join user.roles roles where roles.project.id = :projectId")
     List<Subject> findAllByProjectId(@Param("projectId") Long projectId);
 
-    @Query("select subject from Subject subject left join fetch subject.sources where subject.id =:id")
+    @Query("select subject from Subject subject left join fetch subject.sources "
+        + "where subject.id =:id")
```

Comment:
I think it is not necessary since `subject` has a `oneToOne` relationship between `user`. It should be automatically handled by the framework. In the above query it was necessary to have it explicitly. We are using this method for `api/subjects/{id}` and it is retrieving necessary data. So i think it is not necessary to have it written. WDYT?